typespec-extension/src/code-model-builder.ts

```diff
@@ -677,34 +670,22 @@ export class CodeModelBuilder {
    * @returns filtered api-versions
    */
   private getFilteredApiVersions(
-    pinnedApiVersion: Version | undefined,
-    versions: Version[],
+    pinnedApiVersion: string | undefined,
+    versions: string[],
     excludePreview: boolean = false,
-  ): Version[] {
+  ): string[] {
     if (!pinnedApiVersion) {
       return versions;
     }
     return versions
       .slice(0, versions.indexOf(pinnedApiVersion) + 1)
-      .filter((version) => !excludePreview || !isStable(pinnedApiVersion) || isStable(version));
-  }
-
-  /**
-   * `@armProviderNamespace` currently will add a default server if not defined globally:
-   * https://github.com/Azure/typespec-azure/blob/8b8d7c05f168d9305a09691c4fedcb88f4a57652/packages/typespec-azure-resource-manager/src/namespace.ts#L121-L128
-   * TODO: if the synthesized server has the right hostParameter, we can use that instead
-   *
-   * @param server returned by getServers
-   * @returns whether it's synthesized by `@armProviderNamespace`
-   */
-  private isArmSynthesizedServer(server: HttpServer): boolean {
-    return this.isArm() && (!server.parameters || server.parameters.size == 0);
+      .filter((version) => !excludePreview || pinnedApiVersion.includes("preview") || !version.includes("preview"));
```

Comment:
We are not using TCGC for this filering?

Reply:
We are using TCGC returned api-versions and apply our emitter options to do filtering.

Reply:
Why we no longer use `isStable` func? It seems to be a good abstraction.

BTW, we should actually check `.toLowerCase().includes("-preview")`.

Reply:
Yeah, we should use `isStable` to not duplicate the logic of checking for `-preview` string and also should be case insensitive.